dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java

```diff
@@ -533,6 +533,42 @@ public Long execute(final Jedis client, final ConnectionContext state) throws Dy
         });
     }
 
+    
+    @Override
+    public ScanResult<Map.Entry<String, String>> hscan(final String key, final int cursor) {
```

Comment:
Good points.

I would prefer composition over inheritance for this case. Rather than extending ScanResult, we would create a parameterized interface in dyno-core as well as an implementation class. The implementation class would contain a private reference to ScanResult.